### PR TITLE
Stepangranat/new tolgee core

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

**Comment by Marr11317** [comment]
May I ask how the new api will be?

**Comment by stepan662** [comment]
Not completely established yet. I want to keep integrations API mostly the same. 
Core will definitely have breaking changes as the current API is not ideal and we didn't documented it anywhere, because we knew it was not final.

**Comment by ecstrema** [comment]
Wow this is getting closer and closer to a release!
Congrats!

**Review by cyyynthia — COMMENTED** [review]
Just noting some nitpicks I noticed while exploring the code for things that were unclear in the docs 😅

File: packages/react/src/useTranslateInternal.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    ...defaultOptions,
+    ...options,
+  };
+
+  // dummy state to enable re-rendering
+  const [instance, setInstance] = useState(0);
+
+  const forceRerender = useCallback(() => {
+    setInstance((v) => v + 1);
+  }, [setInstance]);
```
[the remark this answers is not in the document]
Comment (stepan662) [reply packages/react/src/useTranslateInternal.ts:32]: Good point :)

File: packages/react/src/useTranslateInternal.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const t = useCallback(
+    (props: TranslateProps<any>) => {
+      const fallbackNs = props.ns || namespaces?.[0];
+      subscribeToKey({ key: props.key, ns: fallbackNs });
+      return tolgee.t({ ...props, ns: fallbackNs }) as any;
+    },
+    [tolgee, instance]
+  );
+
+  if (currentOptions?.useSuspense && !isLoaded) {
```
[the remark this answers is not in the document]
Comment (stepan662) [reply packages/react/src/useTranslateInternal.ts]: 👍

**Comment by stepan662** [comment]
We'll continue development in [tolgee-5](https://github.com/tolgee/tolgee-js/tree/tolgee-5)

**Pull request merged**